sdk/src/loom/ModestMaps/core/TileGrid.ls

```diff
@@ -0,0 +1,1457 @@
+            var brC:Coordinate = bottomRightCoordinate.zoomTo(currentTileZoom);
+            var trC:Coordinate = topRightCoordinate.zoomTo(currentTileZoom);
+            var blC:Coordinate = bottomLeftCoordinate.zoomTo(currentTileZoom);
+            
+            // optionally pad it out a little bit more with a tile buffer
+            // TODO: investigate giving a directional bias to TILE_BUFFER when panning quickly
+            // NB:- I'm pretty sure these calculations are accurate enough that using 
+            //      Math.ceil for the maxCols will load one column too many -- Tom
+            var minCol:int = Math.floor(Math.min(tlC.column,brC.column,trC.column,blC.column)) - tileBuffer;
+            var maxCol:int = Math.floor(Math.max(tlC.column,brC.column,trC.column,blC.column)) + tileBuffer;
+            var minRow:int = Math.floor(Math.min(tlC.row,brC.row,trC.row,blC.row)) - tileBuffer;
+            var maxRow:int = Math.floor(Math.max(tlC.row,brC.row,trC.row,blC.row)) + tileBuffer;
+
+            
+            // loop over all tiles and find parent or child tiles from cache to compensate for unloaded tiles:
+            repopulateVisibleTiles(minCol, maxCol, minRow, maxRow);
+            
+            // move visible tiles to the end of recentlySeen if we're done loading them
+            // the 'least recently seen' tiles will be removed from the tileCache below
+            for each (var visibleTile:Tile in visibleTiles) {
+                if (tilePainter.isPainted(visibleTile)) {
+                    recentlySeen.remove(visibleTile.name); 
+                    recentlySeen.push(visibleTile.name);
+                }
+            }
+
+            // prune tiles from the well if they shouldn't be there (not currently in visibleTiles)
+            // TODO: unless they're fading in or out?
+            // (loop backwards so removal doesn't change i)
+            for (var i:int = well.numChildren-1; i >= 0; i--) {
+                var wellTile:Tile = well.getChildAt(i) as Tile;
+                if (visibleTiles.indexOf(wellTile) < 0) {
+                    well.removeChild(wellTile);
+                    wellTile.hide();
+                    tilePainter.cancelPainting(wellTile);
+                }
+            }
+
+            // position tiles such that currentZoom is approximately scale 1
+            // and x and y make sense in pixels relative to tlC.column and tlC.row (topleft)
+            positionTiles(tlC.column, tlC.row);
+
+            // all the visible tiles will be at the end of recentlySeen
+            // let's make sure we keep them around:
+            var maxRecentlySeen:int = Math.max(visibleTiles.length, maxTilesToKeep);
+            
+            // prune cache of already seen tiles if it's getting too big:
+            if (recentlySeen.length > maxRecentlySeen) {
+
+                // can we sort so that biggest zoom levels get removed first, without removing currently visible tiles?
+                /*
```

Comment:
There are a lot of sections of code like this inside of the original AS3 port... I'm not sure if the person who did that port just commented out stuff that they didn't need/didn't work, or if it is even like that in the root MMaps source code.

I can go through and nuke them all, but I thought they might need to be around in case we run into expected functionality not working, as they might lead to hints as to why.

Lemme know :)